tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -242,6 +242,55 @@ impl EnvFilter {
     }
 
     fn from_directives(directives: impl IntoIterator<Item = Directive>) -> Self {
+        use tracing::level_filters::STATIC_MAX_LEVEL;
+        use tracing::Level;
+
+        let directives: Vec<_> = directives.into_iter().collect();
+
+        let disabled: Vec<_> = directives
+            .iter()
+            .filter(|directive| directive.level > STATIC_MAX_LEVEL)
+            .collect();
+
+        if !disabled.is_empty() {
+            eprintln!("warning: some trace filter directives would enable traces that are disabled statically");
+            for directive in disabled {
+                let target = if let Some(target) = &directive.target {
+                    format!("the `{}` target", target)
+                } else {
+                    "all targets".into()
+                };
+                let level = directive
+                    .level
+                    .clone()
+                    .into_level()
+                    .expect("=off would not have enabled any filters");
+                eprintln!(
+                    " | `{}` would enable the {} level for {}",
+                    directive, level, target
+                );
+            }
+            eprintln!(" | the static max level is {}", STATIC_MAX_LEVEL);
+            let help_msg = || {
+                let (feature, filter) = match STATIC_MAX_LEVEL.into_level() {
+                    Some(Level::TRACE) => unreachable!(
+                        "if the max level is trace, no static filtering features are enabled"
+                    ),
+                    Some(Level::DEBUG) => ("max_level_debug", Level::TRACE),
+                    Some(Level::INFO) => ("max_level_info", Level::DEBUG),
+                    Some(Level::WARN) => ("max_level_warn", Level::INFO),
+                    Some(Level::ERROR) => ("max_level_error", Level::WARN),
+                    None => return ("max_level_off", String::new()),
+                };
+                (feature, format!("{} ", filter))
+            };
+            let (feature, earlier_level) = help_msg();
+            eprintln!(
+                "note: to enable {}logging, remove the `{}` feature",
+                earlier_level, feature
+            );
```

Comment:
What do you think?

![colors](https://user-images.githubusercontent.com/23638587/95149056-effd4e00-0752-11eb-9fe2-5ef684829d7d.png)

Reply:
For context, this is what rustc's diagnostics look like:

![rustc](https://user-images.githubusercontent.com/23638587/95149097-073c3b80-0753-11eb-90d2-925d557ac5da.png)

Reply:
> i was trying to imitate the formatting used by rustc's error messages, since I know this feature was originally requested for rustc.

I should probably work on colorizing bootstraps errors ... right now they're pretty inconsistent.

```
(-bash@build-server) ~/rustc $ rg println src/bootstrap/
src/bootstrap/config.rs
565:                    println!("failed to parse TOML configuration '{}': {}", file.display(), err);

src/bootstrap/setup.rs
81:                println!("error: unrecognized option '{}'", input.trim());
82:                println!("note: press Ctrl+C to exit");
```

Reply:
> What do you think?
> 
> ![colors](https://user-images.githubusercontent.com/23638587/95149056-effd4e00-0752-11eb-9fe2-5ef684829d7d.png)

this is beautiful <3

Reply:
i _might_ format "the static max level is XXX" as a `note:` but take it or leave it.

Reply:
![note](https://user-images.githubusercontent.com/23638587/95149848-ed9bf380-0754-11eb-955a-f8ca281e3560.png)

Reply:
this looks rad!